### ISSUE TITLE
Invisible player after dead ?

### ISSUE THREAD

**Issue by gacus007** [Nukkit/Nukkit#479]
When player was dead ,he respawn and is invisible - it is correct ?

How stop invisble after dead in plugin player.setVisible ?

**Comment by xxFlare** [comment]
@gacus007 Nukkit should do this, so it may be a bug. You should not need to fix this with the API.

How did this player "die" so I can reproduce this?

**Comment by gacus007** [comment]
Player die from killer's damages ,xxFlare do you have skype or email?

**Comment by gacus007** [comment]
at cn.nukkit.Server.start(Server.java:736)
        at cn.nukkit.Server.<init>(Server.java:440)
        at cn.nukkit.Nukkit.main(Nukkit.java:65)
13:52:39 [NOTICE] Blocked 46.174.4.143 for 5 seconds
13:52:44 [NOTICE] Unblocked 46.174.4.143

**Comment by xxFlare** [comment]
@gacus007 I don't check skype much but I do check my email & twitter. xflare123@gmail.com

**Comment by gacus007** [comment]
Thx, I send you email with errors on email

**Comment by MrPowerGamerBR** [comment]
You could also send the errors in this issue.
Em 11 de fev de 2016 10:03, "John" <notifications@github.com> escreveu:

**Comment by gacus007** [comment]
Ok no problem , pastebin contains 2 errors and main calss my plugin http://pastebin.com/jqPUqcEA

**Comment by MrPowerGamerBR** [comment]
@gacus007 I'm having the same error when using EntityDamageByEntityEvent. "argument type mismatch", maybe @xxFlare or @MagicDroidX can take a look.

On my testings, it seems that this error is caused when the source of the damage is a fire (I'm still testing what causes this error), using "instaceof Player" doesn't work, for some reason.

If needed I can share my EntityDamageByEntityEvent error too :smile:

**Comment by MrPowerGamerBR** [comment]
Well, I decided sharing my logs now instead of waiting for someone to ask for it.

```
2016-2-2 20:46:59 [CRITICAL] Não foi possível passar o evento 'cn.nukkit.event.entity.EntityDamageEvent' para 'SparklyWorldGuard v1.0.0': argument type mismatch em com.mrpowergamerbr.sparklyworldguard.listeners.MoveListener
2016-2-2 20:46:59 [ALERT] java.lang.IllegalArgumentException: argument type mismatch
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at cn.nukkit.plugin.MethodEventExecutor.execute(MethodEventExecutor.java:24)
	at cn.nukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:50)
	at cn.nukkit.plugin.PluginManager.callEvent(PluginManager.java:553)
	at cn.nukkit.entity.Entity.attack(Entity.java:651)
	at cn.nukkit.entity.EntityLiving.attack(EntityLiving.java:107)
	at cn.nukkit.Player.attack(Player.java:3471)
	at cn.nukkit.entity.Entity.entityBaseTick(Entity.java:863)
	at cn.nukkit.entity.EntityLiving.entityBaseTick(EntityLiving.java:184)
	at cn.nukkit.Player.onUpdate(Player.java:1400)
	at cn.nukkit.level.Level.doTick(Level.java:606)
	at cn.nukkit.Server.checkTickUpdates(Server.java:892)
	at cn.nukkit.Server.tick(Server.java:954)
	at cn.nukkit.Server.tickProcessor(Server.java:757)
	at cn.nukkit.Server.start(Server.java:737)
	at cn.nukkit.Server.<init>(Server.java:441)
	at cn.nukkit.Nukkit.main(Nukkit.java:65)
```

The code has a "if (e.getEntity() instanceof Player && e.getEntity().getDamager() instanceof Player) {", so it isn't a missing instance verification.

**Comment by gacus007** [comment]
Could you give some tip for it ?

**Comment by MrPowerGamerBR** [comment]
@gacus007 well, I don't have the issue of the invisible player, but I have the same error as you, maybe they are related...

I think it would be better to create a new issue about the mismatch error.

**Comment by gacus007** [comment]
So sum up api is not complete perfectley ? Thanks a lot for you time for my noobs code ;p,do you have any email?

**Comment by MrPowerGamerBR** [comment]
@gacus007 ...maybe it is a API fault, but I'm not sure, but the error you are getting IS from some error in the API.

**Comment by gacus007** [comment]
Thx a lot and I creating new issue

**Comment by MrPowerGamerBR** [comment]
@gacus007 I already created, you should keep this issue because it is (maybe) a complete unrelated issue

**Comment by gacus007** [comment]
ok ,so we must wait for staff ;p

**Comment by MagicDroidX** [comment]
Can you share your code that you try to listen the event?
